File: frontend/projects/api/src/lib/code-tour-step.ts
export interface CodeTourStep {
  order: number;
  title: string;
  guidance: string;
  repositoryId: string;
  path: string;
  branch?: string;
  commit?: string;
  language: string;
  startLine: number;
  endLine: number;
  symbol: string;
  excerpt: string;
  contentClassification: string;
  provenance: string;
}

File: backend/src/RepoFluent.Application/CodeTourStep.cs
namespace RepoFluent.Application;

public sealed record CodeTourStep(
    int Order,
    string Title,
    string Guidance,
    string RepositoryId,
    string Path,
    string? Branch,
    string? Commit,
    string Language,
    int StartLine,
    int EndLine,
    string Symbol,
    string Excerpt,
    string ContentClassification,
    string Provenance);
